controllers/internalrequest/adapter.go

```diff
@@ -110,6 +119,21 @@ func (a *Adapter) EnsurePipelineRunIsCreated() (reconciler.OperationResult, erro
 	return reconciler.ContinueProcessing()
 }
 
+// EnsurePipelineRunIsDeleted is an operation that will ensure that the PipelineRun created to handle the InternalRequest
+// is deleted once it finishes.
+func (a *Adapter) EnsurePipelineRunIsDeleted() (reconciler.OperationResult, error) {
+	if !a.internalRequest.HasCompleted() || a.internalServicesConfig.Spec.Debug {
+		return reconciler.ContinueProcessing()
```

Comment:
I would add a log message here saying we are not deleting...but maybe in a follow-on PR?

Reply:
Added a log line here.